.github/workflows/bImage.yml

```diff
@@ -0,0 +1,35 @@
+name: "bImage"
+on:
+  push:
+    branches:
+      - main
+      - dev
+      - githactions
+  pull_request:
+    branches:
+      - main
+      - dev
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2.4.0
+    - uses: cachix/install-nix-action@v16
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+    - run: nix build ".#dockerImage"
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME}}
+        password: ${{ secrets.DOCKER_PASSWORD}}
+    - name: push docker image to DockerHub
+      run: |
+        docker load < result
+        docker tag result  personalgamification/github_action
+        docker push personalgamification/github_action:latest
```

Comment:
This isn't going to work since it's trying to push to user/repository so in this case it's trying to push to a user called personalgamification and a repo called github_action. Since you do not own personalgamification this push will not work.